### PR TITLE
Read volatility report and add tests

### PR DESCRIPTION
**Pull request by aarontp**
- Reads volatility report into evidence and Turbinia reporting
- Remove unused debug log
- Also don't need to re-add evidence since the execute() method will add it for us.

### CODE REVIEW

**Comment by aarontp** [comment]
@mwatkins-fb FYI

File: turbinia/workers/volatility.py [hunk 1]
```diff
@@ -60,18 +59,14 @@ def run(self, evidence, result):
...
-        close=True)
+    res = self.execute(cmd, result, new_evidence=[output_evidence], close=True)
 
     if res == 0:
-      # Write the report to the output file.
-      with open(output_file_path, 'w') as fh:
-        fh.write(output_evidence.text_data.encode('utf-8'))
+      # Get report from the output file.
+      with open(output_file_path, 'r') as fh:
+        output_evidence.text_data = fh.read().decode('utf-8')
```
Comment (joachimmetz) [inline turbinia/workers/volatility.py]: What if file_size > 16G ?

Comment (aarontp) [reply turbinia/workers/volatility.py]: This is just for the report text, not the memory image, so I'm not too worried (usually they are just a few kb).

Comment (joachimmetz) [reply turbinia/workers/volatility.py]: If you don't control the size of the file I opt to check, otherwise this could invoke the OOM killer

Comment (aarontp) [reply turbinia/workers/volatility.py]: @joachimmetz  PTAL -- I added a max size that the report will be truncated to if it is too large.  I also added some tests.

File: turbinia/workers/volatility.py [hunk 2]
```diff
@@ -59,23 +60,40 @@ def run(self, evidence, result):
...
-        fh.write(output_evidence.text_data.encode('utf-8'))
-
-      # Add the resulting evidence to the result object.
-      result.add_evidence(output_evidence, evidence.config)
-      result.report_data = output_evidence
-      summary = 'Volatility module {0} successfully ran.'.format(self.module)
+      # Get report data from the output file.
+      try:
+        file_size = os.stat(output_file_path).st_size
+      except OSError as exception:
```
Comment (joachimmetz) [inline turbinia/workers/volatility.py]: I opt: `except (IOError, OSError) as exception:`

File: turbinia/workers/volatility.py [hunk 3]
```diff
@@ -59,23 +60,40 @@ def run(self, evidence, result):
...
+        msg = 'Cannot read output file {0:s}: {1!s}'.format(
```
Comment (joachimmetz) [inline turbinia/workers/volatility.py]: 'Cannot read output file' => 'Unable to determine size of output file'

File: turbinia/workers/volatility.py [hunk 4]
```diff
@@ -59,23 +60,40 @@ def run(self, evidence, result):
...
-      with open(output_file_path, 'w') as fh:
-        fh.write(output_evidence.text_data.encode('utf-8'))
-
-      # Add the resulting evidence to the result object.
-      result.add_evidence(output_evidence, evidence.config)
-      result.report_data = output_evidence
-      summary = 'Volatility module {0} successfully ran.'.format(self.module)
+      # Get report data from the output file.
+      try:
+        file_size = os.stat(output_file_path).st_size
```
Comment (joachimmetz) [inline turbinia/workers/volatility.py]: Note that stat can fail on some cloud file systems

File: turbinia/workers/volatility.py [hunk 5]
```diff
@@ -59,23 +60,40 @@ def run(self, evidence, result):
...
+            'size ({1:d}). Truncating report to max size'.format(
+                file_size, MAX_REPORT_SIZE))
+        summary = (
+            'Volatility module {0:s} successfuly ran (report truncated)'.format(
+                self.module))
+      else:
+        summary = 'Volatility module {0:s} successfully ran'.format(self.module)
+
+      with open(output_file_path, 'rb') as fh:
+        output_evidence.text_data = codecs.decode(
```
Comment (joachimmetz) [inline turbinia/workers/volatility.py]: I opt to now do decode and read on the same line this makes debugging harder
```
report_data = fh.read(MAX_REPORT_SIZE)
output_evidence.text_data = report_data.decode('utf-8')
```
also there is no need to use codecs and make sure to catch UnicodeDecodeErrors

File: turbinia/workers/volatility.py [hunk 6]
```diff
@@ -59,23 +60,40 @@ def run(self, evidence, result):
...
+            'Volatility report output size ({0:d}) is greater than max report '
+            'size ({1:d}). Truncating report to max size'.format(
+                file_size, MAX_REPORT_SIZE))
+        summary = (
+            'Volatility module {0:s} successfuly ran (report truncated)'.format(
+                self.module))
+      else:
+        summary = 'Volatility module {0:s} successfully ran'.format(self.module)
+
+      with open(output_file_path, 'rb') as fh:
```
Comment (joachimmetz) [inline turbinia/workers/volatility.py]: Why not the following approach without stat

first read fh.read(MAX_REPORT_SIZE)
then read 1 byte if len(report_data) == MAX_REPORT_SIZE, if that byte contains data provide the truncation warning

File: turbinia/workers/volatility.py [hunk 7]
```diff
@@ -16,18 +16,21 @@
...
 
 from turbinia import config
 from turbinia.evidence import VolatilityReport
 from turbinia.workers import TurbiniaTask
 
+MAX_REPORT_SIZE = 2**30  # 1 GiB
+
 
 class VolatilityTask(TurbiniaTask):
   """Task to execute volatility."""
```
Comment (joachimmetz) [inline turbinia/workers/volatility.py]: mabye add:

```
Attributes:

  module (TYPE): what module is used for.
```

File: turbinia/workers/volatility.py (hunk 5 above)
Comment (joachimmetz) [reply turbinia/workers/volatility.py]: or alternatively open with encoding (python 3 only) of io.open with encoding (python 2 and 3 compatible)

```
with io.open(output_file_path, 'r', encoding='utf-8') as fh:
  output_evidence.text_data = fh.read(MAX_REPORT_SIZE)
```

File: turbinia/workers/volatility.py (hunk 2 above)
Comment (aarontp) [reply turbinia/workers/volatility.py]: Done.

File: turbinia/workers/volatility.py (hunk 4 above)
Comment (aarontp) [reply turbinia/workers/volatility.py]: This should always be a local directory, so I think we should be fine.

File: turbinia/workers/volatility.py (hunk 7 above)
Comment (aarontp) [reply turbinia/workers/volatility.py]: Done.

File: turbinia/workers/volatility.py (hunk 3 above)
Comment (aarontp) [reply turbinia/workers/volatility.py]: Done.

File: turbinia/workers/volatility.py (hunk 5 above)
Comment (aarontp) [reply turbinia/workers/volatility.py]: Ok -- I split this into two lines, but I didn't use io.open() for now.  Once python2 is completely removed (planned for the release after next), I'll take a pass and upgrade some things like this.

File: turbinia/workers/volatility.py (hunk 6 above)
Comment (aarontp) [reply turbinia/workers/volatility.py]: I'm not sure if I see much of an advantage doing it that way.  Unless there's a compelling reason, I think I'll keep it the way it is for now.

**Comment by aarontp** [comment]
PTAL, thanks!

File: turbinia/workers/volatility.py (hunk 6 above)
Comment (joachimmetz) [reply turbinia/workers/volatility.py]: this was more relevant, when you cannot use stat

**Comment by aarontp** [comment]
> LGTM from my perspective, note that codefactor complains about class member consistency

Thanks for the review.  Yeah, let me know if you have ideas on what the best way to resolve that codefactor issue would be.  I'll merge this for now, and if you have ideas on how to fix that I'll do it in a separate PR since that comes up a few times in the codebase.

**Pull request merged**